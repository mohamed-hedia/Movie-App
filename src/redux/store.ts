import { configureStore } from "@reduxjs/toolkit";
import homeSlice from "./home/homeSlice";
import moviesSlice from "./movies/moviesSlice";
import tvSlice from "./tv/tvSlice";
import detailsSlice from "./details/detailsSlice";
import searchSlice from "./search/searchSlice";
import userSlice from "./user/userSlice";
import bookmarkSlice from "./bookmarked/bookmarkSlice";

export const store = configureStore({
  reducer: {
    user: userSlice,
    home: homeSlice,
    movies: moviesSlice,
    tv: tvSlice,
    details: detailsSlice,
    search: searchSlice,
    bookmark: bookmarkSlice,
  },
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({
      serializableCheck: false,
    }),
});

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;
