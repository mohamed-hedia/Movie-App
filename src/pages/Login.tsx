import { useEffect, useState } from "react";
import { useDispatch, useSelector } from "react-redux";
import { IoIosCheckmarkCircleOutline } from "react-icons/io";
import { useNavigate } from "react-router-dom";

import { AppDispatch, RootState } from "@/redux/store";
import { userLogin } from "@/redux/user/userSlice";
import Button from "@/components/ui/Button";
import Heading from "@/components/ui/Heading";
import Text from "@/components/ui/Text";

const Login = () => {
  const [isLogged, setIsLogged] = useState(false);
  const navigate = useNavigate();
  const { loading, session_id } = useSelector(
    (state: RootState) => state.user
  );

  const dispatch = useDispatch<AppDispatch>();

  const handleLogin = () => {
    dispatch(userLogin());
  };

  const handleNavigate = () => {
    setTimeout(() => navigate("/", { replace: true }), 3000);
  };

  useEffect(() => {
    if (localStorage.getItem("session_id")) {
      setIsLogged(true);
      handleNavigate();
    } else if (session_id) {
      localStorage.setItem("session_id", session_id);
      setIsLogged(true);
      handleNavigate();
    }
  }, [isLogged, session_id]);

  return (
    <div className="w-screen h-screen px-6 md:px-0 flex justify-center items-center">

      {loading && (
        <div className="bg-secondary-dark w-[500px] p-10 flex flex-col justify-center items-center gap-6 rounded-xl">
          <div className="w-14 h-14 border-[5px] border-t-orange rounded-full border-[#ffffff90] animate-spin" />
          <Heading as="h1" className="mt-0">Logging in...</Heading>
        </div>
      )}

      {!loading && !isLogged && (
        <div className="bg-secondary-dark w-[500px] p-10 flex flex-col justify-center gap-6 rounded-xl">
          <Heading as="h1" className="-mt-1" data-test-id='login-heading'>
            Login
          </Heading>
          <Text>
            Please click the login button,
            <br />
            To create a new session for you.
          </Text>
          <Button data-test-id='login-button' className="w-full py-4" onClick={handleLogin}>
            Login
          </Button>
        </div>
      )}

      {isLogged && loading === false ? (
        <div className="bg-secondary-dark w-[500px] p-10 flex flex-col justify-center gap-6 rounded-xl">
          <div className="flex flex-col items-center justify-center gap-6">
            <IoIosCheckmarkCircleOutline className="text-6xl text-orange" />
            <Heading as="h1" className="text-center">
              You are now logged in.
            </Heading>
          </div>
        </div>
      ) : null}
    </div>
  );
};
export default Login;
