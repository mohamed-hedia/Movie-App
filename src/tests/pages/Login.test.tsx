import Login from "@/pages/Login";
import { store } from "@/redux/store";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import { BrowserRouter as Router } from "react-router-dom";


describe("Login Page", () => {
  test("renders Login Page should match the screenshot", () => {
    const login = render(
      <Provider store={store}>
        <Router>
          <Login />
        </Router>
      </Provider>
    );

    expect(login).toMatchSnapshot();
  });
});
