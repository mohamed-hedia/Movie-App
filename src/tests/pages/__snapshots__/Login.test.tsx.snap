// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Login Page > renders Login Page should match the screenshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="w-screen h-screen px-6 md:px-0 flex justify-center items-center"
      >
        <div
          class="bg-secondary-dark w-[500px] p-10 flex flex-col justify-center gap-6 rounded-xl"
        >
          <h1
            class="antialiased text-3xl font-outfitLight -mt-1"
          >
            Login
          </h1>
          <p
            class="antialiased text-base"
          >
            Please click the login button,
            <br />
            To create a new session for you.
          </p>
          <button
            class="bg-orange hover:bg-white px-2 hover:text-black rounded-md text-base w-full py-4"
          >
            Login
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="w-screen h-screen px-6 md:px-0 flex justify-center items-center"
    >
      <div
        class="bg-secondary-dark w-[500px] p-10 flex flex-col justify-center gap-6 rounded-xl"
      >
        <h1
          class="antialiased text-3xl font-outfitLight -mt-1"
        >
          Login
        </h1>
        <p
          class="antialiased text-base"
        >
          Please click the login button,
          <br />
          To create a new session for you.
        </p>
        <button
          class="bg-orange hover:bg-white px-2 hover:text-black rounded-md text-base w-full py-4"
        >
          Login
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
